docs/guide/custom-functions.md

```diff
@@ -46,25 +71,79 @@ import { FunctionPlugin } from 'hyperformula';
       isVolatile: true,
+      repeatLastArgs: 4,
+      vectorizationForbidden: false,
+    }
+  };
+}
+```
+
+You can set the following optional parameters:
+
+| Option | Type | Description |
+| --- | --- | --- |
+| `arrayFunction` | Boolean | If set to `true`, the function enables the [array arithmetic mode](arrays.md) in its arguments and nested expressions. |
+| `doesNotNeedArgumentsToBeComputed` | Boolean | If set to `true`, the function treats reference or range arguments as arguments that don't create dependency.<br><br>Other arguments are properly evaluated. |
+| `expandRanges` | Boolean | If set to `true`, ranges in the function's arguments are inlined to (possibly multiple) scalar arguments. |
+| `isDependentOnSheetStructureChange` | Boolean | If set to `true`, the function gets recalculated with each sheet shape change. |
+| `isVolatile` | Boolean | If set to `true`, the function is [volatile](volatile-functions.md). |
+| `repeatLastArgs` | Number | For functions with a variable number of arguments: sets how many last arguments can be repeated indefinitely. |
+| `vectorizationForbidden` | Boolean | If set to `true`, prevents the function from ever being vectorized (but it's up to your function implementation to properly handle vectorization). |
+
+### Argument validation options
+
+In an optional `parameters` object, you can set rules for your function's argument validation.
+
+```javascript
+// import FunctionPlugin
+import { FunctionPlugin } from 'hyperformula';
+
+// start creating a class
+class CountHF extends FunctionPlugin {
+
+// define functions inside this plugin
+  public static implementedFunctions = {
+    'HYPER': {
+      // this method's functionality will be defined below
+      method: 'hyper',
+      // set your argument validation options below
+      parameters: {
+        passSubtype: false,
+        defaultValue: 10,
+        optionalArg: false,
+        minValue: 5,
+        maxValue: 15,
+        lessThan: 15,
+        greaterThan: 5
+      },
     }
   };
 }
 ```
 
-Similarly, there are other useful properties. 
-`isDependentOnSheetStructureChange` marks functions that need to be recalculated with 
-each change of the shape of the engine sheets. 
-`doesNotNeedArgumentsToBeComputed` marks functions that treat references or ranges in their arguments
-as arguments that do not create dependency. Other arguments are properly evaluated.
-`arrayFunction` denotes functions that enable array arithmetic in its arguments and nested expressions.
-`vectorizationForbidden` when set prevents function from ever being vectorized (however, it is up to implementation of a function to properly handle vectorization).
+You can set the following argument validation options:
+
+| Parameter | Type | Description |
+| --- | --- | --- |
+| `passSubtype` | Boolean | If set to `true`, arguments need to be passed with full type information.<br>(e.g. for numbers: `Date` or `DateTime` or `Time` or `Currency` or `Percentage`) |
```

Comment:
I believe they don't "need to", more like they are. It's a setting for the evaluator to provide them / it means we're requesting types @izulin ?

Reply:
Yes, this is handled by the evaluator. So external function with this flag set to true will receive wrapped numbers instead of just numbers.

Reply:
@wojciechczerniak - updated: f023154